tools/sof-tplgreader.py

```diff
@@ -182,8 +182,12 @@ def _filterOutput(self, pipelines, filter_dict, bIn):
                 if 'any' in value or value == ['']:
                     check = True if key in line.keys() else False
                 else:
-                    # check if current pipeline is among the ones we want
-                    check = str(line[key]) in value
+                    # Use full match for numerical values, and use substring
+                    # match for string values
+                    if str(line[key]).isdigit():
+                        check = str(line[key]) in value
```

Comment:
So  let's take my  previous exemple of an "HDMI42" pipeline and "42" filter. When I read this in the help below:
```
 -f "type:playback"` -> playback pipelines (value substring match)
 -f "id:3"` -> pipeline whose id is 3 (value full match)
```
I understand that it will be a full match because `'42'.isdigit()` is `True`. Now that's very clear with the examples, thanks! But here you seem to test `"HDMI42".isdigit()`?

Reply:
for a pipeline, there are id, name, and other key:value pairs. some value is numerical (id, rate) and others are string(pcm, eq, pga), but the problem is that there is no way to know the value is numerical or not. 

the previous RE is not good, because we cannot tell the different between `id:1` and `id:10`, ce58aed remove RE and use full match, which is not good either, because sometimes, we need to exclude a type of pipelines, for example, HDMI pipelines. the example, `HDMI42` is in pcm kv pair, which should be regarded as string value, btw, 42 is not an id for the pipeline, if we want to match id, use `id:42,43,44`

The 'if all bit in a string is digit, it is numerical value, others are string value' is just a strategy, which is good for value type identification.

Reply:
I don't think you understood my (relatively simple) comment:
- According to the help page, `isdigit()` is run _on the filter_
- When I read the code, I understand that `isdigit()` is run _on the pipelines_

Which is it?

Reply:
Finally, I get the point, thanks. help message updated.